scheduler/scheduler.go

```diff
@@ -152,3 +157,69 @@ func merge(envs ...map[string]string) map[string]string {
 	}
 	return merged
 }
+
+type Status struct {
+	// A friendly human readable message about the status change.
+	Message string
+}
+
+// String implements the fmt.Stringer interface.
+func (s *Status) String() string {
+	return s.Message
+}
+
+// StatusStream is an interface for publishing status updates while a scheduler
+// is executing.
+type StatusStream interface {
```

Comment:
I was thinking some more about this last night and I think there's one more small change we might wanna do that will make this much simpler. We can give semantic meaning to when the StatusStream is nil, and change the signature of `scheduler.Submit` to:

```go
// When StatusStream is nil, Submit should return as quickly as possible, usually when the new version has been received, and validated. If StatusStream is not nil, it's recommended that the method not return until the deployment has fully completed.
scheduler.Submit(context.Context, *App, StatusStream) error
```

This would mean, that we can remove the `Done`, `Wait`, and `Err` methods on the `StatusStream` interface so it's just `Publish`. For the CloudFormation backend, we'd just change [this line](https://github.com/remind101/empire/blob/bd469460a1fc75b387bc49cdc686e05e896bb9d4/scheduler/cloudformation/cloudformation.go#L271)  to be `if ss != nil`.

The primary advantage will be in cases like the [migration scheduler](https://github.com/remind101/empire/blob/bd469460a1fc75b387bc49cdc686e05e896bb9d4/scheduler/cloudformation/migration.go#L152-L163) where we'd call Submit on two different schedulers, that might both call `Done`.